src/cpp/src/llm_pipeline_static.cpp

```diff
@@ -721,6 +721,26 @@ StatefulLLMPipeline::StatefulLLMPipeline(
         ModelConfigDesc model_desc = get_modeldesc_from_json(models_path / "config.json");
         ov::AnyMap properties = config;
         auto compiled = setupAndCompileModel(model, model_desc, properties);
+        // Also export compiled model if required
+        const auto export_blob = pop_or_default(properties, "EXPORT_BLOB", false);
+        if (export_blob) {
+            auto blob_path = pop_or_default(properties, "BLOB_PATH", std::string{});
+            if (blob_path.empty()) {
+                blob_path = (models_path / "openvino_model.blob").string();
+            }
+            // Check the path is full
+            if (blob_path.size() < 5) {
+                OPENVINO_THROW("Please provide a full path to blob file in BLOB_PATH: " + blob_path);
+            }
+            if (strncmp(".blob", &blob_path[blob_path.size() - 5], 5) != 0) {
```

Comment:
Why not `std::string::compare` btw?

Reply:
Maybe this can help: https://en.cppreference.com/w/cpp/filesystem/path/extension

Reply:
I took it from OV util. Couldn't include it here